ntbs-service/wwwroot/source/Components/AutocompleteSelect.ts

```diff
@@ -0,0 +1,65 @@
+import Vue from "vue";
+var accessible = require("accessible-autocomplete");
+
+const AutocompleteSelect = Vue.extend({
+    props: ["placeholder", "validate"],
+    mounted: function() {
+        var selectElement = this.$el;
+        var selectId = selectElement.id;
+        accessible.enhanceSelectElement({
+            placeholder: this.$props.placeholder,
+            autoselect: false,
+            preserveNullOptions: true,
+            showAllValues: true,
+            selectElement: selectElement,
+            onConfirm: (val: any) => {
+                this.onConfirm(val);
+            }
+        })
+        this.autocompleteElement = document.getElementById(selectId) as HTMLInputElement;
+        this.selectElement = document.getElementById(this.autocompleteElement.id + "-select") as HTMLSelectElement;
+        // Workaround for stopping Chrome trying to autocomplete by country, see discussion https://gist.github.com/niksumeiko/360164708c3b326bd1c8
+        // Note that autocomplete="off" does not work.
+        this.autocompleteElement.setAttribute('autocomplete', 'no');
+        // As we do not build the element ourselves, use ordinary event listeners rather than vue's
+        if (this.$props.validate) {
+            this.autocompleteElement.addEventListener('change', (event: any) => this.inputChanged(event));
+            this.selectElement.addEventListener('select-changed', (event: any) => this.selectChanged(event));
+        }
+    },
+    methods: {
+        inputChanged: function (event: FocusEvent) {
+            if (this.autocompleteElement.value === "")
+            {
+                // Autocomplete does not automatically select blank value when deleting text rather than clicking on blank, so emit this manually
+                this.selectElement.value = "";
+                this.$emit("validate-input", event);
+            }
+        },
+        onConfirm: function (value: any) {
+            // this is called twice, value is undefined when user clicks away from input field which is when we want validation to happen
+            if (!this.$props.validate || value != undefined) {
+                return;
+            }
+            var event = new Event('select-changed');
+            var autocompleteValue = this.autocompleteElement.value;
+            // the actual select element does not have its value set at this point; the only way we seem to be able to find it is by searching for the matching autocomplete

```

Comment:
surely it does get the value changed at some point - after all it will be sent as part of the form submission?
Are we not able to identify the time when it does get it changed? would a listener on the `select` element not be possible?

Reply:
in fact, it seems like you're doing that change by yourself in the loop below - surely the component has that behaviour by itself though, no?

Reply:
At the point the change listener on the autocomplete is fired, the value of the original select field is still the old one.... I would have expected this not to be the case, and don't quite understand when it gets updated (because before I did any of that it definitely submitted the correct value after form submission), it is really weird...
I could try a short timeout within the change listener in case there is a delay to this being set?

Reply:
There's a chance it may have been updated in the virtual dom, but not been rendered to screen yet?

Reply:
@conorsheehansw is there a way to access the virtual dom?

Reply:
@jasiekmiko I just tried having a long timeout after onConfirm to see whether the value of the original select field had been updated, but it's always stuck on the value before the latest change. I really don't understand when the change happens and what to listen out for to make this work. None of the inbuilt event handlers on the original select (change, input, blur) ever fire. Without spending a lot more time on this, not sure whether I can make this work without the manual workaround

Reply:
RE: Virtual DOM not as far as I'm aware. 
You can use nextTick with a callback, or forceUpdate (I do not have experience of using these though...): https://vuejs.org/v2/api/#vm-forceUpdate